src/components/timeline/form/TimelineDetailsPanel.svelte

```diff
@@ -124,130 +264,398 @@
+                  </span>
+                  {row.name}
+                  <div>
+                    <button
+                      use:tooltip={{ content: 'Edit Row', placement: 'top' }}
+                      class="st-button icon"
+                      on:click={() => {
+                        viewSetSelectedRow(row.id);
+                      }}
+                    >
+                      <PenIcon />
+                    </button>
+                    <button
+                      use:tooltip={{ content: 'Delete Row', placement: 'top' }}
+                      class="st-button icon"
+                      on:click|stopPropagation={() => handleDeleteRowClick(row)}
+                    >
+                      <TrashIcon />
+                    </button>
+                  </div>
+                </div>
+              </div>
+            {/each}
+          </div>
+        </fieldset>
+      {/if}
+    {:else}
+      <!-- Row editing -->
+      <button
+        on:click={() => {
+          viewSetSelectedRow(null);
         }}
+        class="st-button tertiary section-back-button"
       >
-        {#each $view.definition.plan.timelines as timeline}
-          <option value={timeline.id}>
-            Timeline {timeline.id}
-          </option>
-        {/each}
-      </select>
-    </div>
-
-    <!-- Timeline -->
-    {#if !$selectedTimeline}
-      <fieldset class="editor-section">No timeline selected</fieldset>
-    {:else}
+        <ArrowLeftIcon />
+        Back to Timeline {$selectedTimelineId}
+      </button>
+      <div class="timeline-select-container">
+        <select
+          class="st-select w-100"
+          data-type="number"
+          name="rows"
+          value={$selectedRowId}
+          on:change={e => {
+            const { valueAsNumber: id } = getTarget(e);
+            viewSetSelectedRow(id);
+          }}
+        >
+          {#each rows as row (row.id)}
+            <option value={row.id}>
+              {row.name}
+            </option>
+          {/each}
+        </select>
+      </div>
       <fieldset class="editor-section">
-        <div class="st-typography-medium editor-section-header">Margins</div>
-        <CssGrid columns="1fr 1fr" gap="8px">
+        <div class="st-typography-medium editor-section-header">Details</div>
+        <div style="display: grid">
           <Input>
-            <label for="marginLeft">Margin Left</label>
+            <label for="name">Row Name</label>
             <input
               class="st-input w-100"
-              name="marginLeft"
-              type="number"
-              value={$selectedTimeline.marginLeft}
-              on:input={updateTimelineEvent}
+              name="name"
+              autocomplete="off"
+              type="string"
+              value={$selectedRow.name}
+              on:input|stopPropagation={updateRowEvent}
             />
           </Input>
-
+        </div>
+        <CssGrid columns="1fr 1fr" gap="8px">
```

Comment:
The 'row height' section alignment here is slightly off:
<img width="473" alt="Screen Shot 2022-12-15 at 1 13 17 PM" src="https://user-images.githubusercontent.com/683355/207968333-0dc11b84-3c19-45a0-b693-6fa4a6fb937f.png">

Reply:
oh weird, is this in some non-Chrome browser?

Reply:
Don't see it in any of my browsers yet

Reply:
Nope saw it on Chrome - weird let me check again

Reply:
This is what i see in Chrome Version 108.0.5359.124 (Official Build) (arm64)
<img width="300" alt="image" src="https://user-images.githubusercontent.com/4419607/207983363-b68e31e0-d3e6-4231-a6e2-56da947f43e6.png">

Reply:
What happens if you add it to the right-most panel?

Reply:
Nothing different though i wonder if i should start fresh and see if the issue is seen during default view?
<img width="1315" alt="image" src="https://user-images.githubusercontent.com/4419607/207984752-cf6ba50c-579b-4510-aa64-35e22f14550e.png">

Reply:
I don't see the issue anymore locally 👀 Not sure. Call it fixed! Must have been something on my end.

Reply:
That is super weird. Will keep an eye out for it when i work on the layers section..